### PR TITLE
fix the bug of connection in nuvolos

### PR DESCRIPTION
**Pull request by zhangzcjames**

### CODE REVIEW

File: R/get_connection.R [hunk 1]
```diff
@@ -199,3 +125,77 @@ credd_from_local <- function(){
...
+  } 
+  
+  # Using Kube secret files to substitute.
+  if (is.null(username) && is.null(password)) {
+    
+    path_user <- '/secrets/username'
+    path_token <- '/secrets/snowflake_access_token'
+    
+    if (file.exists(path_user) && file.exists(path_token)) {
+      print("Using secret files.")
```
Comment (andrewsali) [inline R/get_connection.R]: I don't think we need this, not sure if the user will understand if this has any implication.

File: R/get_connection.R [hunk 2]
```diff
@@ -199,3 +125,77 @@ credd_from_local <- function(){
...
+      con_user <- file(path_user, "r")
+      line_user <- readLines(con_user, n = 1, warn = FALSE)
+      close(con_user)
+      
+      if( length(line_user) == 0)
+        stop(paste0('Could not parse username file, first line of ', path_user, ' is empty.'))
```
Comment (andrewsali) [inline R/get_connection.R]: I always prefer having explicit { }

File: R/get_connection.R [hunk 3]
```diff
@@ -199,3 +125,77 @@ credd_from_local <- function(){
...
+      close(con_pw)
+      
+      if (length(line_pw) == 0 )
+        stop(paste0('Could not parse token file, first line of ', path_token ,' is empty.'))
+      password <- line_pw
+    } else {
+      stop("Cannot find correct credential...")
+    }
+  }
+  return(list(username, password, dbname, schemaname))
```
Comment (andrewsali) [inline R/get_connection.R]: I would prefer a named list

File: R/get_connection.R [hunk 4]
```diff
@@ -17,91 +17,22 @@
...
-      password = cred[2]
-    }
-    
-    
-  } else if(is.null(username) && !is.null(password) ) {
-    stop("Inconsistent input: username is NULL, but password was provided. Please specify a username or leave both arguments as NULL.")
-  } else if(!is.null(username) && is.null(password) ) {
-    stop("Inconsistent input: password is NULL, but username was provided. Please specify a password or leave both arguments as NULL.")
-  }
+  username = conn_param[[1]]
```
Comment (andrewsali) [inline R/get_connection.R]: Please use a named list to not have to use numeric indexing

File: R/get_connection.R [hunk 5]
```diff
@@ -199,3 +125,77 @@ credd_from_local <- function(){
...
+  return(list(username, password, dbname, schemaname))
+}
+
+get_nuvolos_info <- function(username = NULL, password = NULL, dbname = NULL, schemaname = NULL){
+  # Relying on get_nuvolos_db_path() to find db and schema. 
+  if (is.null(dbname) && is.null(schemaname)) {
+    conn_info <- get_nuvolos_db_path()
+    dbname <- conn_info$dbname
+    schemaname <- conn_info$schemaname
+  }
```
Comment (andrewsali) [inline R/get_connection.R]: we should have an else if (is.null(dbname) ||  is.null(schemaname)) to make sure that if at least one is specified, then both are (similarly like for local connector)

**Review by andrewsali — CHANGES_REQUESTED** [review]

File: R/get_connection.R (hunk 1 above)
Comment (zhangzcjames) [reply R/get_connection.R]: done, deleted this

File: R/get_connection.R (hunk 2 above)
Comment (zhangzcjames) [reply R/get_connection.R]: done

File: R/get_connection.R (hunk 3 above)
Comment (zhangzcjames) [reply R/get_connection.R]: yep, good point, done.

**Pull request merged**